src/Uno.UI/UI/Xaml/Input/PointerRoutedEventArgs.macOS.cs

```diff
@@ -23,22 +22,83 @@ internal PointerRoutedEventArgs(NSSet touches, NSEvent nativeEvent) : this()
 			_nativeEvent = nativeEvent;
 			_nativeTouches = touches;
 
-			FrameId = _pseudoFrameId;
-			Pointer = new Pointer(nativeEvent);
+			var pointerId = (uint)0;
+			var pointerDeviceType = GetPointerDeviceType(nativeEvent.Type);
+
+			var isInContact = GetIsInContact(nativeEvent);
+
+			FrameId = ToFrameId(_nativeEvent.Timestamp);
+			Pointer = new Pointer(pointerId, pointerDeviceType, isInContact, isInRange: true);
+			KeyModifiers = System.VirtualKeyModifiers.None; //TODO: Properly set virtual key modifiers
+			OriginalSource = null; //TODO: get original source
 			CanBubbleNatively = true;
 		}
 
-
 		public PointerPoint GetCurrentPoint(UIElement relativeTo)
 		{
 			var device = PointerDevice.For(PointerDeviceType.Mouse);
-			var point = relativeTo.ConvertPointFromView(_nativeEvent.LocationInWindow, null);
-			var properties = new PointerPointProperties() { IsInRange = true, IsPrimary = true };
+			var rawPosition = _nativeEvent.LocationInWindow;
+			var position = relativeTo != null ?
+				relativeTo.ConvertPointFromView(_nativeEvent.LocationInWindow, null) :
+				rawPosition;
+			
+			var properties = new PointerPointProperties()
+			{
+				IsInRange = true,
+				IsPrimary = true,
+				IsLeftButtonPressed = ((int)NSEvent.CurrentPressedMouseButtons & LeftMouseButtonMask) == LeftMouseButtonMask,
```

Comment:
@dr1rrb Regarding the touch input, I think there is no support for touch in macOS at all, so it would be only guesswork currently. Maybe I am just not aware of something though :-D .

Reply:
I would have agree, until I noticed that there is also [a `TouchesBegan` on macOS ](https://developer.apple.com/documentation/appkit/nsresponder/1531151-touchesbegan)

Maybe for the touch pad ?

Anyway, we as didn't override those methods, I would assume that we will get only mouse pointers, so you can disregard my comments about pen and fingers (maybe just add a comment to remind us that we should probably have to updates those code once we have overridden all the input methods)

Reply:
Hmm, one thing which could it work with is when you use iPad as second screen. I will check that 🤔

Reply:
Ok, so I managed to improve support for pen with the iPad Sidecar for some of the events to return pressure and tilt, however it seems like it is all in a "war state" - for example pen events are in fact reported as mouse events, for some the NSEvent.Subtype, which should work for mouse events throws for MouseEntered and MouseExited (so it is impossible to detect whether the mouse event is in fact a pen event) and so on. Finally, the Sidecar does not support touch input, so I was unable to implement that, yet. However, the mouse events are now working nicely, button click and button pressed state too, after applying your suggestions :-)

Reply:
At some point I'm happy to not be the only one which had to deal with this nightmare that is unifying the mouse/pen/touch handling on all platforms ... but trust me, I understand your feeling right now ;)